packages/core/src/modules/attendance/attendee-service.ts

```
@@ -0,0 +1,139 @@
+}
+
+export class AttendeeServiceImpl implements AttendeeService {
+  constructor(
+    private readonly attendeeRepository: AttendeeRepository,
+    private readonly attendancePoolRepository: AttendancePoolRepository,
+    private readonly attendanceRespository: AttendanceRepository,
+    private readonly userService: UserService
+  ) {
+    this.attendeeRepository = attendeeRepository
+    this.userService = userService
+  }
+
+  async create(obj: AttendeeWrite) {
+    return this.attendeeRepository.create(obj)
+  }
+
+  async delete(id: AttendeeId) {
+    await this.attendeeRepository.delete(id)
+  }
+
+  async getByUserId(userId: UserId, attendanceId: AttendanceId) {
+    const attendee = await this.attendeeRepository.getByUserId(userId, attendanceId)
+    return attendee
+  }
+
+  async updateAttended(attended: boolean, id: AttendeeId) {
+    const attendee = await this.attendeeRepository.update({ attended }, id)
+    const user = await this.userService.getUserById(attendee.userId)
+
+    if (user === undefined) {
+      throw new Error("User not found")
+    }
+
+    return {
+      ...attendee,
+      user,
+    }
+  }
+
+  async updateExtraChoices(id: AttendanceId, questionId: string, choiceId: string) {
+    const attendee = await this.attendeeRepository.updateExtraChoices(id, questionId, choiceId)
+
+    if (attendee === null) {
+      throw new Error("Attendee not found, this should not happen seeing that it was just updated.")
+    }
+    return attendee
+  }
+
+  async registerForEvent(userId: UserId, attendanceId: AttendanceId, now: Date) {
+    const user = await this.userService.getUserById(userId)
+
+    if (user === undefined) {
+      throw new Error("User not found")
+    }
+
+    // is user already registered?
+    const userAlreadyRegistered = await this.attendeeRepository.getByUserId(userId, attendanceId)
+    if (userAlreadyRegistered !== null) {
+      throw new Error("User is already registered")
+    }
```

Maybe this should silent fail?